firebolt_provider/hooks/firebolt.py

```diff
@@ -188,25 +170,17 @@ def engine_action(self, engine_name: Optional[str], action: str) -> None:
                 "engine_name is not set, getting engine_name from connection config"
             )
             conn_config = self._get_conn_params()
-            database_name = conn_config.get("database")
-
             engine_name = conn_config.get("engine_name", None)
-            engine_url = conn_config.get("engine_url", None)
-            if engine_url:
-                raise FireboltError("Cannot start/stop engine using its URL")
+            if engine_name is None:
+                raise FireboltError("Engine name must be provided")
 
         rm = self.get_resource_manager()
-        if engine_name is None:
-            if database_name is None:
-                raise FireboltError("Database must be set")
-            engine = get_default_database_engine(rm, database_name)
-        else:
-            engine = rm.engines.get_by_name(engine_name)
+        engine = rm.engines.get(engine_name)
 
         if action == "start":
-            engine.start(wait_for_startup=True)
+            engine.start()
         elif action == "stop":
-            engine.stop(wait_for_stop=True)
+            engine.stop()
```

Comment:
For me this function should be separated: `retrieve_eingine(engine_name)` and `run_action(engine, action: str)`.
In this  case function `engine_action()` could be re-written as 
```python
 def engine_action(self, engine_name: Optional[str], action: str) -> None:
    run_action(retrieve_engine(engine_name), action)
```